cpu/stm32f1/periph/gpio.c

```diff
@@ -98,15 +98,27 @@ int gpio_init(gpio_t pin, gpio_mode_t mode)
     }
 #endif
 
+#ifdef BOARD_BLUEPILL
```

Comment:
Why not name this `STM32L1_SWJ_TO_GPIO` as suggested and define it in `periph_conf.h` for the board?

Also `gpio_init()` should fail for those pins if this is not set, right?

Reply:
For the sake of this PR simplicity I suggested to leave it board specific, as I think that if we introduce this we should document it properly. Also we should introduce `STM32F1_SWJ_RST_TO_GPIO` to remove the nucleo board (although setting `AFIO_MAPR_SWJ_CFG_DISABLE` also changes the SWJ RST, this would give more granularity to the configuration), and as you mention change the return value.

If @jn-powerlan has the time and is willing to do this then better, if not I offer to do the follow-up myself.

Reply:
@leandrolanzieri i am agree to do the modification, it's a good trainning to participate at RIOT.